.github/workflows/tests.yml

```diff
@@ -19,22 +19,52 @@ jobs:
 
     steps:
     - uses: actions/checkout@v1
+
     - name: Set up Python 3.6
       uses: actions/setup-python@v1
       with:
         python-version: 3.6
+
+    - name: Pip Cache
+      uses: actions/cache@v1
+      id: pip-cache
+      with:
+        path: ~/.cache/pip
+        key: ${{ runner.os }}-pip-${{ hashFiles('Pipfile.lock') }}
+        restore-keys: ${{ runner.os }}-pip-
+
+    - name: Pipenv cache
+      uses: actions/cache@v1
+      id: pipenv-cache
+      with:
+        path: ~/.cache/pipenv
+        key: ${{ runner.os }}-pipenv-${{ hashFiles('Pipfile.lock') }}
+        restore-keys: ${{ runner.os }}-pipenv-
+
+    - name: Virtualenv cache
+      uses: actions/cache@v1
+      id: virtualenv-cache
+      with:
+        path: ~/.local/share/virtualenvs
+        key: ${{ runner.os }}-virtualenvs-${{ hashFiles('Pipfile.lock') }}
```

Comment:
Changing the entire virtual env every time a package is update seems a bit counter intuitive, it should only need to be rebuilt if we upgrade python version, right?

Reply:
Yeah, I agree, but this is at least better than rebuilding it for every git push. Maybe it is possible to change this key to be something like `${{ runner.os }}-virtualenvs-3.6` or something like that, and then bump it to 3.7 when we eventually upgrade.

I might do something like that in a while. Thanks for pointing it out. :)